sdk/formrecognizer/Azure.AI.FormRecognizer/tests/Models/FieldValueTests.cs

```diff
@@ -17,5 +18,88 @@ public void AsStringReturnsNullWhenFieldValueIsDefault()
             FieldValue value = default;
             Assert.IsNull(value.AsString());
         }
+
+        [Test]
+        public void InstantiateFieldValueWithFloatAndNoValueNumberAndText()
+        {
+            var fieldValueInternal = new FieldValue_internal(
+                FieldValueType.Float,
+                valueString: default,
+                valueDate: default,
+                valueTime: default,
+                valuePhoneNumber: default,
+                valueNumber: default,
+                valueInteger: default,
+                valueArray: default,
+                valueObject: default,
+                valueSelectionMark: default,
+                valueGender: default,
+                valueCountry: default,
+                text: "25.00%",
+                boundingBox: default,
+                confidence:  default,
+                elements: default,
+                page: default);
+
+            var fieldValue = new FieldValue(fieldValueInternal, default);
+
+            Assert.AreEqual(FieldValueType.Float, fieldValue.ValueType);
+            Assert.Throws<InvalidOperationException>(() => fieldValue.AsFloat());
+        }
+
+        [Test]
+        public void InstantiateFieldValueWithInt64AndNoValueInteger()
+        {
+            var fieldValue = new FieldValue(FieldValueType.Int64);
+
+            Assert.AreEqual(FieldValueType.Int64, fieldValue.ValueType);
+            Assert.Throws<InvalidOperationException>(() => fieldValue.AsInt64());
+        }
+
+        [Test]
+        public void InstantiateFieldValueWithFloatAndNoValueNumber()
+        {
+            var fieldValue = new FieldValue(FieldValueType.Float);
+
+            Assert.AreEqual(FieldValueType.Float, fieldValue.ValueType);
+            Assert.Throws<NullReferenceException>(() => fieldValue.AsFloat());
```

Comment:
We shouldn't be throwing a `NullReferenceException` here. This is happening because we're doing
```cs
if (float.TryParse(_fieldValue.Text.TrimStart('$'), out float parsedFloat))
```
without a null check:
```cs
if (_fieldValue.Text != null && float.TryParse(_fieldValue.Text.TrimStart('$'), out float parsedFloat))
```

I can understand why this is happening in this test, but can `Text` be null in actual service calls?

Reply:
>can Text be null in actual service calls?

No. As it is what the service uses first for recognition. If the service couldn't recognize the text and map to a field, then the service won't return the field.

As for the test, I kind of prefer to have it like this so if it changes we can easily understand what is happening. Having said that, I am also ok if you prefer for me to add the null check and have a specific check that will throw if text is empty

Reply:
No strong feelings about it. I just think I might get a bit confused if I stumble upon this test again in the future.

Reply:
I added a comment that explains this so in the future if someone sees this method, hopefully, it won't block them